src/som/interpreter/actors/EventualMessage.java

```diff
@@ -323,24 +331,40 @@ public static EventualMessage getCurrentExecutingMessage() {
 
   /**
    * Indicates that execution should stop and yield to the debugger,
-   * before the computed value is used to resolve the promise.
+   * before the message is processed.
+   */
+  public boolean haltOnReceive() {
+    return haltOnReceive;
```

Comment:
Could you be more specific why this is confusing?
`isHaltOnReceive` doesn't read well, "is halt" isn't really English.
Personally, I don't see this as an issue, but then again, I have used languages where it is more common to name accessors like the fields they access.